app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,35 @@
+class PiecesController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    @piece = current_game.pieces.create(piece_params)
+  end
+
+  def show
+    @piece = Piece.find(params[:id])
+  end
+
+  def update
+    row = params[:row]
+    column = params[:column]
+    piece = Piece.find(params[:id])
+    piece.move_to!(column, row)
```

Comment:
Looks good! Could reduce this a bit by doing the following.

```Ruby
piece = Piece.find(params[:id])
piece.move_to!(params[:column], params[:row])
```

Reply:
Thanks, I will change it for the next task!